packages/devextreme/testing/tests/DevExpress.ui.widgets/mapParts/azureTests.js

```diff
@@ -0,0 +1,884 @@
+        const done = assert.async();
+
+        if(window.atlas) {
+            delete window.atlas;
+        }
+
+        const d1 = $.Deferred();
+        const d2 = $.Deferred();
+
+        $('<div>').appendTo($('#map')).dxMap({
+            provider: 'azure',
+            onReady: $.proxy((e) => {
+                assert.ok(window.atlas, 'map loaded');
+
+                d1.resolve();
+            }, this)
+        });
+
+        $('<div>').appendTo($('#map')).dxMap({
+            provider: 'azure',
+            onReady: $.proxy((e) => {
+                assert.ok(window.atlas, 'map loaded');
+
+                d2.resolve();
+            }, this)
+        });
+
+        $.when(d1, d2).done(function() {
+            done();
+        });
+    });
+
+    QUnit.test('map ready action', function(assert) {
+        const done = assert.async();
+
+        $('#map').dxMap({
+            provider: 'azure',
+            onReady: (e) => {
+                assert.ok(true, 'map ready');
+                assert.strictEqual(atlas.optionsSpecified, true, 'map options specified');
+                assert.ok(e.originalMap instanceof atlas.Map, 'map instance specified');
+
+                done();
+            }
+        });
+    });
+});
+
+QUnit.module('basic options', moduleConfig, () => {
+    ['someKey', { azure: 'someKey' }].forEach((apiKey) => {
+        QUnit.test(`map credintials should be passed to a config (apiKey is a ${typeof apiKey})`, function(assert) {
+            const done = assert.async();
+
+            $('#map').dxMap({
+                provider: 'azure',
+                apiKey,
+                onReady: () => {
+                    const authOptions = {
+                        authType: 'subscriptionKey',
+                        subscriptionKey: 'someKey',
+                    };
+                    assert.deepEqual(atlas.options.authOptions, authOptions, 'map credentials are passed to config');
+
+                    done();
+                }
+            });
+        });
+    });
+
+    [
+        {
+            type: 'roadmap',
+            style: 'road',
+        },
+        {
+            type: 'hybrid',
+            style: 'satellite_road_labels',
+        },
+        {
+            type: 'satellite',
+            style: 'satellite',
+        }
+    ].forEach(({ type, style }) => {
+        QUnit.test(`map should have style=${style} when type option=${type}`, function(assert) {
+            const done = assert.async();
+
+            $('#map').dxMap({
+                provider: 'azure',
+                type,
+                onReady: () => {
+                    assert.strictEqual(atlas.options.style, style, 'style is passed correctly');
+
+                    done();
+                }
+            });
+        });
+    });
+
+    QUnit.test('map should change style on runtime type update', function(assert) {
+        const done = assert.async();
+        const d = $.Deferred();
```

Comment:
[code style]
Let's update everywhere
```suggestion
        const mapReadyDeferred = $.Deferred();
```

Reply:
updated everywhere except for cases with 2-3 defferred